app/javascript/components/visualization/ScatterPlot.js

```diff
@@ -94,6 +101,34 @@ function RawScatterPlot({
     return scatter
   }
 
+  /** Save any changes to the legend colors */
+  async function saveCustomColors(newColors) {
+    const colorObj = {}
+    colorObj[annotation.name] = newColors
```

Comment:
I've noted that if you're making an edit on the default annotation, this gets set to `""`, when it should either be `_default`, or ideally the actual name of the default annotation.